integrationtests/git/git_test.go

```diff
@@ -0,0 +1,374 @@
+				t.Errorf("latestCommit doesn't match. got %s, expected %s", latestCommit, test.expectedCommit)
+			}
+		})
+	}
+}
+
+func TestLatestCommitSSH(t *testing.T) {
+	ctx := context.Background()
+	container, url, err := createGogsContainer(ctx, createTempFolder(t))
+	if err != nil {
+		t.Errorf("got error when none was expected: %v", err)
+	}
+	defer func() {
+		if err := container.Terminate(ctx); err != nil {
+			t.Fatalf("failed to terminate container: %s", err.Error())
+		}
+	}()
+	privateKey, err := createAndAddKeys(url)
+	if err != nil {
+		t.Errorf("got error when none was expected: %v", err)
+	}
+	sshPort, err := container.MappedPort(ctx, "22")
+	if err != nil {
+		t.Errorf("got error when none was expected: %v", err)
+	}
+	gogsKnownHosts := []byte("[localhost]:" + sshPort.Port() + " " + gogsFingerPrint)
+
+	tests := map[string]struct {
+		gitjob         *gitjobv1.GitJob
+		knownHosts     []byte
+		expectedCommit string
+		expectedErr    error
+	}{
+		"public repo": {
+			gitjob: &gitjobv1.GitJob{
+				Spec: gitjobv1.GitJobSpec{
+					Git: gitjobv1.GitInfo{
+						Repo:   "ssh://git@localhost:" + sshPort.Port() + "/test/" + "public-repo",
+						Branch: "master",
+					},
+				},
+			},
+			knownHosts:     gogsKnownHosts,
+			expectedCommit: latestCommitPublicRepo,
+			expectedErr:    nil,
+		},
+		"private repo with known hosts": {
+			gitjob: &gitjobv1.GitJob{
+				Spec: gitjobv1.GitJobSpec{
+					Git: gitjobv1.GitInfo{
+						Repo:   "ssh://git@localhost:" + sshPort.Port() + "/test/" + "private-repo",
+						Branch: "master",
+					},
+				},
+			},
+			knownHosts:     gogsKnownHosts,
+			expectedCommit: latestCommitPrivateRepo,
+			expectedErr:    nil,
+		},
+		"private repo without known hosts": {
+			gitjob: &gitjobv1.GitJob{
+				Spec: gitjobv1.GitJobSpec{
+					Git: gitjobv1.GitInfo{
+						Repo:   "ssh://git@localhost:" + sshPort.Port() + "/test/" + "private-repo",
+						Branch: "master",
+					},
+				},
+			},
+			knownHosts:     nil,
+			expectedCommit: latestCommitPrivateRepo,
+			expectedErr:    nil,
+		},
+		"private repo with known host with a wrong host url": {
+			gitjob: &gitjobv1.GitJob{
+				Spec: gitjobv1.GitJobSpec{
+					Git: gitjobv1.GitInfo{
+						Repo:   "ssh://git@localhost:" + sshPort.Port() + "/test/" + "private-repo",
+						Branch: "master",
+					},
+				},
+			},
+			knownHosts:     []byte("doesntexits " + gogsFingerPrint),
+			expectedCommit: "",
+			expectedErr:    fmt.Errorf("ssh: handshake failed: knownhosts: key is unknown"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			secret := &v1.Secret{
+				Data: map[string][]byte{
+					v1.SSHAuthPrivateKey: []byte(privateKey),
+					"known_hosts":        test.knownHosts,
+				},
+				Type: v1.SecretTypeSSHAuth,
+			}
+			secretGetter := &secretGetterMock{secret: secret}
+			latestCommit, err := git.LatestCommit(test.gitjob, secretGetter)
+
+			if !reflect.DeepEqual(err, test.expectedErr) {
```

Comment:
Do we need equality here, or would [errors.Is](https://pkg.go.dev/errors#Is) do the trick? This could make these test cases more flexible if we decide to enhance error handling with prefixing, wrapping errors with more context across calls.

Reply:
unfortunately errors.Is fails even if the errors are the same